account_credit_control/models/credit_control_line.py

```diff
@@ -256,15 +277,68 @@ def write(self, values):
             self.partner_id.write({
                 'manual_followup': values.get('manual_followup'),
             })
+        for line in self:
+            if (
+                'auto_process' not in values and
+                line.policy_id.auto_process_lower_levels
+            ):
+                related_lines = line.get_related_lines()
+                if not related_lines:
+                    return res
+                related_lines[0].write({'auto_process': 'highest_level'})
+                if len(related_lines) > 1:
+                    related_lines[1:].write({'auto_process': 'low_level'})
         return res
 
     @api.model_create_multi
     def create(self, vals_list):
         lines = super(CreditControlLine, self).create(vals_list)
         for line in lines:
             line.manual_followup = line.partner_id.manual_followup
+            if line.policy_id.auto_process_lower_levels:
+                related_lines = line.get_related_lines()
+                if not related_lines:
+                    continue
+                related_lines[0].write({'auto_process': 'highest_level'})
+                if len(related_lines) > 1:
+                    related_lines[1:].write({'auto_process': 'low_level'})
         return lines
 
+    def get_related_lines(self):
+        """
+        Return lines from the same group if grouped
+        (ie with same partner, policy and currency).
+
+        The most important line (ie the one to display to the user)
+        is the first one of the returned recordset.
+        """
+        self.ensure_one()
+        if self.policy_id.auto_process_lower_levels:
+            return self.search(
+                [
+                    ('partner_id', '=', self.partner_id.id),
+                    ('currency_id', '=', self.currency_id.id),
+                    ('policy_id', '=', self.policy_id.id),
+                    ('state', 'in', ('draft', 'to_be_sent')),
+                ],
+                order='level DESC, date_due ASC',
+            )
+        else:
+            return self
+
+    def get_lower_related_lines(self):
+        """
+        Return lines that will receive the same treatment
+        (ie lines of lower level from the same group if grouped).
+        """
+        self.ensure_one()
+        if self.policy_id.auto_process_lower_levels:
+            return self.get_related_lines().filtered(
```

Comment:
a search will be better the a filter.

Reply:
yes, since `get_related_lines` already does a search it will be more efficient to add a level criteria in the search. OTOH it's probably marginal since the number of related lines will never be huge.

Reply:
I added a level criteria and refactored this part a bit.